nodes/monitor.go

```diff
@@ -204,6 +209,45 @@ func (mon *NodeMonitor) doChecks() {
 	}
 }
 
+func (mon *NodeMonitor) provideBadBlocks(r *Report) {
+	for _, block := range r.BadBlocks {
+		fname := fmt.Sprintf("www/badblocks/0x%x.json", block.Hash)
+		// only write it if it isn't already there
+		if _, err := os.Stat(fname); os.IsNotExist(err) {
+			var data []byte
+			// try to retrieve header from backend
+			if bl := mon.backend.get(block.Hash); bl != nil {
+				data, err = json.MarshalIndent(bl, "", " ")
+				if err != nil {
+					log.Warn("Failed to marshall header", "error", err)
+				}
+			}
+			if len(data) == 0 {
+				// block not found in backend, print what we know
```

Comment:
When investigating bad blocks, `rlp` is good to have. And since we already have it in the backed, we might aswell make it accessible to the UI aswell, so people can click the thing and copy-paste the RLP to check it out locally.

Reply:
It took me ages to figure it out, but finally the rlp gets marshalled in correctly